File: 2014/s2/s2.py
#Open file
File = open("s2.in",'r')

#Read the number of students
N = int(File.readline())

#Read the group of students
first = File.readline().split()
second = File.readline().split()

state = "good"
i = 0
#Iterate through names unless state becomes "bad"
while state == "good" and i <= len(first)-1:
	#Find the index in the first list of i's partner
	x = second.index(first[i])
	#Check that i is x's partner and vice versa
	#Check that students are not assigned to themselves
	if (not(first[i] == second[x] and second[i] == first[x])) or (first[i] == second[i]):
		#Throw "bad" flag
		state = "bad"
	i = i +1
print state

File: FirstStage/2014/s2/s2.cpp
#include <stdio.h>
#include <string>
#include <iostream>

using namespace std;

int main(int argc, char *argv[])	
{
	//Read the number of students
	int N = 0;
	scanf("%d",&N);

	//The first line of student names
	string  first[30];
	//The second line of student names
	string second[30];

	//Read the first line of names
	for(int i = 0; i < N; ++i)
	{
		cin >> first[i];
	}
	
	//Read the second line of names
	for(int i = 0; i < N; ++i)
	{
		cin >> second[i];
	}

	//The state of the partner assignment
	string state = "good";

	int i = 0;
	while(i < N && state == "good")
	{
		//The partner of the student in the first line
		string partner = second[i];
		
		//Search first line for partner
		int j = 0;
		while(j < N && first[j] != partner)
		{
			j++;
		}

		//Check if the partner in the first line is not paired up with the original
		//student in the second line
		//or if student is paired with themself
		if(second[j] != first[i] || first[i] == first[j])
		{
			state = "bad";
		}
		i++;
	}

	cout << state << endl;

	return 0;
}
